cadc-vos-server/src/main/java/ca/nrc/cadc/vos/server/transfers/PullFromVOSpaceNegotiation.java

```diff
@@ -98,31 +102,75 @@
     }
 
+    @Override
+    public void validateView() throws IllegalArgumentException,
+            TransferException, Exception {
+        // There is no representation for the package view in View.properties,
+        // so this function needs to do some extra work for this instance.
+        // For all other View values, the standard validation is performed.
+
+        if (TransferUtil.isPackageTransfer(transfer)) {
+            isPackageViewTransfer = true;
+        } else {
+            super.validateView();
+        }
+    }
+
     @Override
     public void doAction()
         throws TransferException, JobPersistenceException, JobNotFoundException,
         LinkingException, NodeNotFoundException, TransferParsingException,
         IOException, TransientException, URISyntaxException {
         boolean updated = false;
         try {
+
+            if (isPackageViewTransfer) {
+                // Set job back to PENDING so next service in the chain can pick it up
+                // (in this case /vault/pkg/<jobid>
+                updateTransferJob(null, null, ExecutionPhase.PENDING);
+
+                // Add responseFormat parameter to the job, using the parameter
+                // type from the view
+                Parameter p = TransferUtil.viewParam2JobParam(VOS.PROPERTY_URI_FORMAT,
+                    "responseFormat", transfer);
+
+                if (p != null) {
+                    // View parameter with uri=VOS.PROPERTY_URI_FORMAT found
+                    List<Parameter> params = new ArrayList<>();
+                    params.add(p);
+
+                    // need job id here, and uws parameter made from transfer view paramter
+                    jobUpdater.addParameters(job.getID(), params);
+                }
+
+                // Set this value so in the finally clause updateTransferJob() is not
+                // run again.
+                updated = true;
+                return;
+            }
+
             // Even though Transfer.java supports multiple targets, this type
-            // of transfer does not yet. When view=zip | tar is implemented,
-            // this is likely to change.
+            // of transfer does not yet.
             // This call confirms a single target exists or throws TransferException.
             TransferUtil.confirmSingleTarget(transfer);
 
             VOSURI target = new VOSURI(transfer.getTargets().get(0));
 
             PathResolver resolver = new PathResolver(nodePersistence);
 
-            // TODO: change this to check all targets provided in the transfer
+            // In future: change this to check all targets provided in the transfer
+            // Check at line 163 ensures this part of the code will only be arrived at
+            // if one target is given. Supporting multiple targets for anything other
+            // than zip|tar packages is considered outside scope
+            // of CADC-10509 - supporting package delivery.
```

Comment:
I find this comment a bit confusing... regular transfers will never have multiple targets, so I think you can delete these comment lines.

Reply:
removed comment